src/utils/txtRecord.ts

```diff
@@ -0,0 +1,32 @@
+export function parseTxtRecord(buffer: ArrayBuffer | Uint8Array): string {
+  const view = buffer instanceof Uint8Array ? buffer : new Uint8Array(buffer);
+  const decoder = new TextDecoder();
+  let result = '';
+  let offset = 0;
+
+  while (offset < view.length) {
+    const length = view[offset];
+    offset += 1;
+
+    if (offset + length > view.length) {
+      throw new Error('Invalid TXT record format: length exceeds buffer size');
+    }
+
+    const slice = view.subarray(offset, offset + length);
+    result += decoder.decode(slice);
+    offset += length;
+  }
+
+  return result;
+}
+
+export function processTxtRecordData(data: unknown): string {
```

Comment:
wondering if we could add some more semantic meaning to these ones - not clear to me when I would use them